CarRecognition/Source Files/Common/Services/CarClassificationService.swift

```diff
@@ -69,16 +81,62 @@ internal final class CarClassificationService {
         self.state = state
     }
     
-    private func handleDetection(request: VNRequest, error: Error?) {
-        guard
-            let results = request.results,
-            let classifications = results as? [VNClassificationObservation]
-        else {
-            return
+    private func performClassification(on pixelBuffer: CVPixelBuffer) {
+        guard state == .running else { return }
+        if type == .detection {
```

Comment:
`if type == .detection && !isReadyForNextFrame { return }` 
Definitely can be done in one line, please just double check if it works.

Reply:
I guess you're right, thanks!